puppet/modules/freight/templates/deploy_debs.erb

```diff
@@ -43,6 +47,17 @@ files.each do |file|
 end
 system "/usr/bin/freight-cache -v -c /home/freight/freight.conf apt/#{os}"
 
+foreman_release_dir = "#{ENV['TARGETCACHE']}/pool/#{os}/#{version}/f/foreman-release/"
+release_files = Dir.glob("#{foreman_release_dir}/foreman-release_*.deb")
+# we assume the file names can be alphabetically sorted to find the newest
+latest_release_file = release_files.sort.last
```

Comment:
Well, this assumption is clearly wrong, I am stupid, and need to think more about that!

```
> ["foreman-release_3.4.0-1_all.deb", "foreman-release_3.4.0~rc1-1_all.deb", "foreman-release_3.4.0~rc2-1_all.deb"].sort.last
=> "foreman-release_3.4.0~rc2-1_all.deb"
```

Reply:
mtime it is!